### PR TITLE
Fix blockquotes display, wrap long lines and refactor + optimize code for future mail processing

### PR DESCRIPTION
**Pull request by LunarX**

### CODE REVIEW

**Review by FabianDevel — CHANGES_REQUESTED** [review]
We could create a WebViewUtils class which can be instantiated as lazy by NewMessageFragment and ThreadAdapter
This class could wrap the 3 lazy and the load/process functions to factorize it

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt [hunk 1]
```diff
@@ -167,17 +175,19 @@ class ThreadAdapter : RecyclerView.Adapter<ThreadViewHolder>(), RealmChangesBind
         var styledBody = if (type == TEXT_PLAIN) createHtmlForPlainText(bodyWebView) else bodyWebView
         styledBody = processMailDisplay(styledBody, uid)
 
+        settings.loadWithOverviewMode = true
+        settings.useWideViewPort = true
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt:179]:
```suggestion
        settings.apply {
            useWideViewPort = true
            loadWithOverviewMode = true
        }
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt (hunk 1 above)
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt:179]: See in later PR

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt [hunk 2]
```diff
@@ -306,10 +313,17 @@ class NewMessageFragment : Fragment() {
...
 
-    private fun WebView.loadContent(html: String) {
-        var content = if (context.isNightModeEnabled()) context.injectCssInHtml(R.raw.custom_dark_mode, html) else html
-        content = context.injectCssInHtml(R.raw.remove_margin, content)
-        loadDataWithBaseURL("", content, ClipDescription.MIMETYPE_TEXT_HTML, Utils.UTF_8, "")
+    private fun WebView.loadContent(html: String) = with(HtmlFormatter(html)) {
+        settings.apply {
+            useWideViewPort = true
+            loadWithOverviewMode = true
+        }
```
[the remark this answers is not in the document]
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt:320]: See in later PR

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt [hunk 3]
```diff
@@ -307,9 +309,13 @@ class NewMessageFragment : Fragment() {
     }
 
     private fun WebView.loadContent(html: String) {
-        var content = if (context.isNightModeEnabled()) context.injectCssInHtml(R.raw.custom_dark_mode, html) else html
-        content = context.injectCssInHtml(R.raw.remove_margin, content)
-        loadDataWithBaseURL("", content, ClipDescription.MIMETYPE_TEXT_HTML, Utils.UTF_8, "")
+        settings.apply {
+            useWideViewPort = true
+            loadWithOverviewMode = true
+        }
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt:315]: This should also be a part of WebViewUtils

File: app/src/main/java/com/infomaniak/mail/utils/WebViewUtils.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    private val customDarkMode by lazy { context.getCustomDarkMode() }
+    private val setMargin by lazy { context.getSetMargin() }
+    private val customStyle by lazy { context.getCustomStyle() }
+
+    fun processHtml(html: String, isDisplayedInDarkMode: Boolean): String = with(HtmlFormatter(html)) {
+        if (isDisplayedInDarkMode) registerCss(customDarkMode, DARK_BACKGROUND_STYLE_ID)
+        registerCss(setMargin)
+        registerCss(customStyle)
+        registerMetaViewPort()
+        return inject()
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/utils/WebViewUtils.kt:35]:
```suggestion
        registerMetaViewPort()

        return inject()
```

**Review by FabianDevel — CHANGES_REQUESTED** [review]

**Review by JorisBodin — APPROVED** [review]

**Review by FabianDevel — APPROVED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**